examples/platform/silabs/BaseApplication.cpp

```diff
@@ -802,6 +802,8 @@ void BaseApplication::ScheduleFactoryReset()
         {
             Provision::Manager::GetInstance().SetProvisionRequired(true);
         }
+        // Delete all fabrics.
+        chip::Server::GetInstance().GetFabricTable().DeleteAllFabrics();
```

Comment:
Omitting this is intentional, see `examples/platform/silabs/BaseApplication.h` line 245:
`   /**
     * @brief Start the factory Reset process
     *  Almost identical to Server::ScheduleFactoryReset()
     *  but doesn't call GetFabricTable().DeleteAllFabrics(); which deletes Key per key.
     *  With our KVS platform implementation this is a lot slower than deleting the whole kvs section
     *  our silabs nvm3 driver which end up being doing in ConfigurationManagerImpl::DoFactoryReset(intptr_t arg).
     */ `

Reply:
Since wifi is using a different nvm3 driver, I suggest using `Server::ScheduleFactoryReset() ` inside an `#idfef` for wifi rather than modifying `BaseApplication::ScheduleFactoryReset()`

Reply:
Added the wifi condition since it is needed only for wifi.Not using Server::ScheduleFactoryReset since we still need the setProvisionRequired event. 
Not adding this in the src/platform/silabs/configurationManager.cpp since it will create a circular dependency by including app in the platform